packages/waku/src/lib/renderers/rsc-renderer.ts

```diff
@@ -97,7 +99,14 @@ export async function renderRsc(
     {
       get(_target, encodedId: string) {
         const [file, name] = encodedId.split('#') as [string, string];
-        const id = resolveClientEntry(file, config);
+        let id = resolveClientEntry(file, config);
+        if (isDev) {
+          for (const moduleNode of opts.initialModules) {
+            if (moduleNode.file === file) {
+              id = moduleNode.url;
+            }
+          }
+        }
```

Comment:
It's better to add DEV-only code here. Can you patch `resolveClientEntry` in the caller?